labcontrol/db/process.py

```diff
@@ -2520,35 +2520,75 @@ def _format_picklist(vol_sample, max_vol_per_well=60000,
         dest_plate_shape: list of 2 elements
             The destination plate shape
         """
+
+        def get_well_name(offset_from_ascii_ucase_a, col_num):
+            row_ascii_code = ord('A') + offset_from_ascii_ucase_a
+            row_letter = chr(row_ascii_code)
+            return "%s%d" % (row_letter, col_num)
+
         if dest_plate_shape is None:
             dest_plate_shape = [16, 24]
 
         contents = ['Source Plate Name,Source Plate Type,Source Well,'
                     'Concentration,Transfer Volume,Destination Plate Name,'
                     'Destination Well']
-        # Write the sample transfer volumes
-        rows, cols = vol_sample.shape
+        num_input_rows, num_input_cols = vol_sample.shape
+        running_tot = 0
+        num_dest_rows = dest_plate_shape[0]
+        num_dest_cols = dest_plate_shape[1]
+        dest_well_index = 0
+
         # replace NaN values with 0s to leave a trail of unpooled wells
         pool_vols = np.nan_to_num(vol_sample)
-        running_tot = 0
-        d = 1
-        for i in range(rows):
-            for j in range(cols):
-                well_name = "%s%d" % (chr(ord('A') + i), j + 1)
-                # Machine will round, so just give it enough info to do the
-                # correct rounding.
-                val = "%.2f" % pool_vols[i][j]
-                # test to see if we will exceed total vol per well
-                if running_tot + pool_vols[i][j] > max_vol_per_well:
-                    d += 1
-                    running_tot = pool_vols[i][j]
+
+        for curr_input_row_index in range(num_input_rows):
+            for curr_input_col_index in range(num_input_cols):
+                curr_input_well_name = get_well_name(curr_input_row_index,
+                                                     curr_input_col_index + 1)
+
+                # test to see if adding the volume of this input well to the
```

Comment:
It looks like the addition is performed on line 2560?

Is this comment in reference to the `if` on line 2554? I think I'm getting tossed up by the proximity of the comment. Would it be possible to modify this comment to something along the lines of "test to see if the current well exceeds the total volume per well. If it does not, then test to see if adding requires incrementing the well index"? 

~~I almost wonder if the check performed on line 2554 would make sense to perform in bulk at or near line 2542?~~ Scratch that, that would complicate providing a user digestible error message.

Reply:
Yeah, I see what you mean.  I will move/tweak the comment.

Reply:
Thanks!!